src/pepr/operator/controllers/network/generators/kubeNodes.ts

```diff
@@ -148,6 +152,78 @@ export async function updateKubeNodesNetworkPolicies() {
   }
 }
 
+/**
+ * Updates the AuthorizationPolicies for KubeNodes.
+ *
+ * This function rebuilds the current set of node peers from the in-memory node set,
+ * extracts their CIDR strings, and then queries for all AuthorizationPolicies that are labeled
+ * with "uds/generated" equal to RemoteGenerated.KubeNodes. For each matching policy, it checks
+ * whether the current IP blocks in the policy's "from" source match the newly computed IP blocks.
+ * If they differ, the policy is updated (with managedFields cleared to avoid server-side apply issues)
+ * and then re-applied.
+ *
+ * @returns {Promise<void>} A promise that resolves once the update process is complete.
+ */
+export async function updateKubeNodesAuthorizationPolicies(): Promise<void> {
+  // Build the current set of node peers from nodeSet.
+  const newPeers = buildNodePolicies([...nodeSet]);
+  // Extract CIDR strings from the new peers.
+  const newIpBlocks = newPeers
+    .map(peer => peer.ipBlock?.cidr)
+    .filter((cidr): cidr is string => typeof cidr === "string");
+
+  // Query for AuthorizationPolicies labeled with uds/generated=KubeNodes.
+  log.debug("Calling K8s(AuthorizationPolicy).WithLabel(...).Get()");
```

Comment:
nit: locally this was kind of noisy in the logs for me, maybe more of a trace log if we need to enable it for dev purposes?